cli/server/server.go

```diff
@@ -530,12 +533,14 @@ Main:
 				log.Info("SIGHUP received, restarting rpc-server")
 				rpcServer.Shutdown()
 				rpcServer = server.New(chain, cfg.ApplicationConfiguration.RPC, serv, oracleSrv, log, errChan)
-				rpcServer.Start()
+				serv.AddService(&rpcServer) // Replaces old one by service name.
+				if !cfg.ApplicationConfiguration.RPC.StartWhenSynchronized || serv.IsInSync() {
+					rpcServer.Start()
+				}
 			}
 		case <-grace.Done():
 			signal.Stop(sighupCh)
 			serv.Shutdown()
-			rpcServer.Shutdown()
```

Comment:
Why is this line removed? I thought `started` check in `shutdown` fixed any possible issues?

Reply:
`network.(*Server)` stops all services.